File: database-driver-plugin/src/main/java/de/lino/database/database/sql/mariadb/MariaDBDatabaseProvider.java
package de.lino.database.database.sql.mariadb;

import de.lino.database.database.auth.Credentials;
import de.lino.database.database.DatabaseType;
import de.lino.database.database.sql.SQLDatabaseProvider;
import de.lino.database.database.sql.SQLExecution;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * The {@link SQLDatabaseProvider} for MariaDB, connected via {@link SQLExecution} using
 * {@link DatabaseType#MARIA_DB}'s JDBC driver and URL scheme.
 */
@Getter
public class MariaDBDatabaseProvider extends SQLDatabaseProvider {

    /**
     * Connects to a MariaDB database with {@code credentials}.
     *
     * @param credentials the login credentials and connection details to connect with
     */
    public MariaDBDatabaseProvider(@NotNull Credentials credentials) {
        super(DatabaseType.MARIA_DB, new SQLExecution(DatabaseType.MARIA_DB, credentials));
    }

}

File: python/database-driver-plugin/src/database_driver/plugin/database/sql/mariadb/mariadb_database_provider.py
"""Mirror of ``de.lino.database.database.sql.mariadb.MariaDBDatabaseProvider``."""

from __future__ import annotations

from database_driver.api.database.auth.credentials import Credentials
from database_driver.api.database.database_type import DatabaseType
from database_driver.plugin.database.sql.sql_database_provider import SQLDatabaseProvider
from database_driver.plugin.database.sql.sql_execution import SQLExecution


class MariaDBDatabaseProvider(SQLDatabaseProvider):
    """The ``SQLDatabaseProvider`` for MariaDB, connected via ``SQLExecution`` using
    ``DatabaseType.MARIA_DB``'s driver and connection scheme."""

    def __init__(self, credentials: Credentials) -> None:
        super().__init__(DatabaseType.MARIA_DB, SQLExecution(DatabaseType.MARIA_DB, credentials))
